lib/core_ext/numpy.rb

```diff
@@ -8,6 +8,22 @@ def d2!
     raise 'not a 2d array' if !first.is_a?(Array)
   end
 
+  # verify points are in the right order and in range.
+  def pts!(r1, c1, r2, c2)
```

Comment:
do we have a better name for this? maybe valid_pts!

Reply:
it could also call a includes_pt?(r,c) helper which would be useful for other reasons